Scoop/View/EditProfileViewController.swift

```diff
@@ -0,0 +1,256 @@
+//
+//  EditProfileViewController.swift
+//  Scoop
+//
+//  Created by Caitlyn Jin on 9/27/23.
+//
+
+import SDWebImage
+import UIKit
+
+class EditProfileViewController: UIViewController {
+    
+    // MARK: - Views
+    
+    private let profileImageView = UIImageView()
+    private let uploadPhotoButton = UIButton()
+    
+    private let stackView = UIStackView()
+    private let nameTextField = UITextField()
+    private let pronounsTextField = UITextField()
+    private let hometownTextField = UITextField()
+    private let classTextField = UITextField()
+    
+    weak var delegate: ProfileViewDelegate?
+    
+    // MARK: - User Data
+    
+    private var user: BaseUser?
+    
+    private var hometown: String?
+    private var talkative: Float?
+    private var music: Float?
+    private var snack: String?
+    private var song: String?
+    private var stop: String?
+    
+    // MARK: - Lifecycle Functions
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = UIColor.white
+        
+        setupHeader()
+        setupProfileImageView()
+        setupUploadPhotoButton()
+        setupStackView()
+    }
+    
+    /// Temporary while save button is not implemented
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        
+        getUserPreferences()
+        updateUser()
+    }
+    
+    // MARK: - Initializers
+    
+    init(user: BaseUser, hometown: String) {
+        super.init(nibName: nil, bundle: nil)
+        self.user = user
+        if let profilePicUrl = user.profilePicUrl {
+            profileImageView.sd_setImage(with: URL(string: profilePicUrl))
+        } else {
+            profileImageView.image = UIImage(named: "emptyimage")
+        }
+        nameTextField.text = "\(user.firstName) \(user.lastName)"
+        pronounsTextField.text = user.pronouns
+        hometownTextField.text = hometown
+        classTextField.text = user.grade
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup View Functions
+    
+    private func setupHeader() {
+        let dottedLineMultiplier = 0.52
+        let solidLineVerticalInset = -12.75
+        let solidLineMultiplier = 0.32
+        let screenSize = UIScreen.main.bounds
+        let dottedline = UIImageView(image: UIImage(named: "dottedline"))
```

Comment:
IDK if Scooped does this, but usually these images should be factored out to a `UIImage+Extension` file. You can simply just do `UIImage.scooped.dottedLine`.